Chapter 5 Gene Ontology Analysis
5.1 Supported organisms

5.2 GO classification


In [1]:
library(clusterProfiler)
data(geneList, package="DOSE")

Warning message:
“multiple methods tables found for ‘splitAsList’”
Warning message:
“replacing previous import ‘S4Vectors::splitAsList’ by ‘IRanges::splitAsList’ when loading ‘AnnotationDbi’”


clusterProfiler v3.17.0  For help: https://guangchuangyu.github.io/software/clusterProfiler

If you use clusterProfiler in published research, please cite:
Guangchuang Yu, Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.


Attaching package: ‘clusterProfiler’


The following object is masked from ‘package:stats’:

    filter




In [2]:
library(org.Hs.eg.db)

Loading required package: AnnotationDbi

Loading required package: stats4

Loading required package: BiocGenerics

Loading required package: parallel


Attaching package: ‘BiocGenerics’


The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB


The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs


The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which, which.max, which.min


Loading required package: Biobase

In [3]:
gene <- names(geneList)[abs(geneList) > 2]
gene.df <- bitr(gene, fromType = "ENTREZID",
                toType = c("ENSEMBL", "SYMBOL"),
                OrgDb = org.Hs.eg.db)
head(gene.df)

'select()' returned 1:many mapping between keys and columns



ENTREZID ENSEMBL         SYMBOL
1 4312     ENSG00000196611 MMP1  
2 8318     ENSG00000093009 CDC45 
3 10874    ENSG00000109255 NMU   
4 55143    ENSG00000134690 CDCA8 
5 55388    ENSG00000065328 MCM10 
6 991      ENSG00000117399 CDC20

In [4]:
ggo <- groupGO(gene     = gene,
               OrgDb    = org.Hs.eg.db,
               ont      = "CC",
               level    = 3,
               readable = TRUE)

In [5]:
head(ggo)

ID         Description                    Count GeneRatio
GO:0005886 GO:0005886 plasma membrane                52    52/207   
GO:0005628 GO:0005628 prospore membrane               0    0/207    
GO:0005789 GO:0005789 endoplasmic reticulum membrane  8    8/207    
GO:0019867 GO:0019867 outer membrane                  2    2/207    
GO:0031090 GO:0031090 organelle membrane             13    13/207   
GO:0034357 GO:0034357 photosynthetic membrane         0    0/207    
           geneID                                                                                                                                                                                                                                                                                                                      
GO:0005886 S100A9/MELK/S100A8/MARCO/ASPM/CXCL10/LAMP3/CEP55/UGT8/UBE2C/SLC7A5/CXCL9/FADS2/MSLN/IL1R2/KIF18A/S100P/GZMB/TRAT1/GABRP/AQP9/GPR19/SLC2A6/KIF20A/LAG3/NUDT1/CACNA1D/VSTM4/CORIN/KCNK15/CA12/KCNE4/HLA-DQA1/ADH1B/PDZK1/C7/ACKR1/COL17A1/PSD3/EMCN/SLC44A4/LRP2/NLGN4X/MAPT/ERBB4/CX3CR1/LAMP5/ABCA8/STEAP4/PTPRT/TMC5/CYBRD1
GO:0005628                                                                                                                                                                                                                                                                                                                             
GO:0005789 FADS2/CDK1/CHODL/ITPR1/HLA-DQA1/CYP4F8/CYP4B1/FMO5                                                                                                                                                                                                                                                                          
GO:0019867 MAOB/PGR                                                                                                                                                                                                                                                                                                                    
GO:0031090 MARCO/LAMP3/DUSP2/DTL/NUDT1/MAOB/ITPR1/GASK1B/HLA-DQA1/LRP2/LAMP5/STEAP4/PGR                                                                                                                                                                                                                                                
GO:0034357

5.3 GO over-representation test


In [6]:
ego <- enrichGO(gene          = gene,
                universe      = names(geneList),
                OrgDb         = org.Hs.eg.db,
                ont           = "CC",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05,
                readable      = TRUE)
head(ego)

ID         Description                              GeneRatio
GO:0000779 GO:0000779 condensed chromosome, centromeric region 14/197   
GO:0000776 GO:0000776 kinetochore                              15/197   
GO:0000775 GO:0000775 chromosome, centromeric region           17/197   
GO:0000793 GO:0000793 condensed chromosome                     17/197   
GO:0005819 GO:0005819 spindle                                  21/197   
GO:0098687 GO:0098687 chromosomal region                       21/197   
           BgRatio   pvalue       p.adjust     qvalue      
GO:0000779 85/11434  1.662431e-10 3.802012e-08 3.572360e-08
GO:0000776 104/11434 2.551686e-10 3.802012e-08 3.572360e-08
GO:0000775 147/11434 5.495085e-10 5.458451e-08 5.128746e-08
GO:0000793 154/11434 1.141189e-09 8.501857e-08 7.988322e-08
GO:0005819 256/11434 2.923725e-09 1.742540e-07 1.637286e-07
GO:0098687 265/11434 5.435150e-09 2.699458e-07 2.536403e-07
           geneID                                                                                                                    
GO:0000779 CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/CDT1/BIRC5/NCAPG/AURKB/CCNB1                                        
GO:0000776 CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/AURKB/CCNB1                                   
GO:0000775 CDCA8/CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/NCAPG/AURKB/CCNB1                       
GO:0000793 CENPE/NDC80/TOP2A/NCAPH/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/CDT1/BIRC5/NCAPG/AURKB/CHEK1/CCNB1                      
GO:0005819 CDC20/KIF23/CENPE/ASPM/DLGAP5/SKA1/TPX2/TACC3/NEK2/CDK1/MAD2L1/KIF18A/KIF11/TTK/AURKB/PRC1/KIF18B/KIF20A/AURKA/CCNB1/KIF4A
GO:0098687 CDCA8/CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/CDK1/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/EZH2/TTK/NCAPG/AURKB/CHEK1/CCNB1/MCM5  
           Count
GO:0000779 14   
GO:0000776 15   
GO:0000775 17   
GO:0000793 17   
GO:0005819 21   
GO:0098687 21

In [7]:
ego2 <- enrichGO(gene         = gene.df$ENSEMBL,
                 OrgDb         = org.Hs.eg.db,
                 keyType       = 'ENSEMBL',
                 ont           = "CC",
                 pAdjustMethod = "BH",
                 pvalueCutoff  = 0.01,
                 qvalueCutoff  = 0.05)

In [8]:
ego2 <- setReadable(ego2, OrgDb = org.Hs.eg.db)
head(ego2)

ID         Description                                GeneRatio
GO:0042613 GO:0042613 MHC class II protein complex               14/228   
GO:0030669 GO:0030669 clathrin-coated endocytic vesicle membrane 14/228   
GO:0000779 GO:0000779 condensed chromosome, centromeric region   15/228   
GO:0000776 GO:0000776 kinetochore                                16/228   
GO:0000775 GO:0000775 chromosome, centromeric region             18/228   
GO:0000793 GO:0000793 condensed chromosome                       18/228   
           BgRatio   pvalue       p.adjust     qvalue      
GO:0042613 95/20630  2.518558e-12 3.312305e-10 2.608882e-10
GO:0030669 96/20630  2.919990e-12 3.312305e-10 2.608882e-10
GO:0000779 119/20630 4.231645e-12 3.312305e-10 2.608882e-10
GO:0000776 142/20630 4.632594e-12 3.312305e-10 2.608882e-10
GO:0000775 202/20630 1.150023e-11 6.578130e-10 5.181155e-10
GO:0000793 209/20630 2.037631e-11 9.712706e-10 7.650051e-10
           geneID                                                                                                                       
GO:0042613 HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1
GO:0030669 HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1/HLA-DQA1
GO:0000779 CENPE/NDC80/HJURP/SKA1/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/CDT1/BIRC5/NCAPG/AURKB/CCNB1                                      
GO:0000776 CENPE/NDC80/HJURP/SKA1/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/AURKB/CCNB1                                 
GO:0000775 CDCA8/CENPE/NDC80/HJURP/SKA1/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/NCAPG/AURKB/CCNB1                     
GO:0000793 CENPE/NDC80/TOP2A/NCAPH/HJURP/SKA1/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/CDT1/BIRC5/NCAPG/AURKB/CHEK1/CCNB1                    
           Count
GO:0042613 14   
GO:0030669 14   
GO:0000779 15   
GO:0000776 16   
GO:0000775 18   
GO:0000793 18

5.3.1 drop specific GO terms or level
5.3.2 test GO at sepcific level
5.3.3 reduce redundancy of enriched GO terms
5.4 GO Gene Set Enrichment Analysis

In [9]:
ego3 <- gseGO(geneList     = geneList,
              OrgDb        = org.Hs.eg.db,
              ont          = "CC",
              nPerm        = 1000,
              minGSSize    = 100,
              maxGSSize    = 500,
              pvalueCutoff = 0.05,
              verbose      = FALSE)

Warning message in .GSEA(geneList = geneList, exponent = exponent, minGSSize = minGSSize, :
“We do not recommend using nPerm parameter incurrent and future releases”
Warning message in fgsea(pathways = geneSets, stats = geneList, nperm = nPerm, minSize = minGSSize, :
“You are trying to run fgseaSimple. It is recommended to use fgseaMultilevel. To run fgseaMultilevel, you need to remove the nperm argument in the fgsea function call.”


5.5 GO Semantic Similarity Analysis

5.5.1 GO analysis for non-model organisms